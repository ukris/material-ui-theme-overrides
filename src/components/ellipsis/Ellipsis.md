
Editable Ellipsis
```jsx
    <Ellipsis
        text="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis vestibulum augue, ut ullamcorper velit. Etiam sodales odio ut luctus efficitur"
        editable={true}
        classes={}
        updateRow={text => alert(`Sent to parent ${text}`)}
    />
```