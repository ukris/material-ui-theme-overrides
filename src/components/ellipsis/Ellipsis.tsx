import React from 'react'
import TextField from '@material-ui/core/TextField'
import clsx from 'clsx'
import './styles.scss'

export interface EllipsisProps {
    /** Default string to be rendered */
    text: string
    classes: any  
    /** Updates row data in parent component with the purpose of persisting when switching table page */
    updateCell: (text: string) => void
}

export default function Ellipsis({text, updateCell, classes}: EllipsisProps) {
  const [value, setValue] = React.useState(text);
  const textfieldRef = React.useRef(null)
  const { fontSize } = classes
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setValue(event.target.value);
  };

  const handleNewInsertedValue = () => updateCell(value)


  React.useEffect(() => {
    // @ts-ignore
    if(textfieldRef && textfieldRef.current && textfieldRef.current.firstChild && textfieldRef.current.firstChild.children[0].nodeName === 'TEXTAREA') {
      // @ts-ignore
      textfieldRef.current.firstChild.children[0].focus()
    }
  })
  
  let cls = 'editable' 
  cls = clsx(cls, 'table-ellipsis', fontSize)

  return(
    <div className="table-ellipsis-container">
      <TextField
        data-test="table-ellipsis"
        ref={textfieldRef}
        className={cls}
        title={undefined}
        id="standard-multiline-flexible"
        multiline={true}
        value={value}
        variant= "outlined"
        onChange={handleChange}
        InputProps={{
          classes: {
            input: fontSize,
          },
        }}
        onBlur={handleNewInsertedValue}
      />
    </div>
  )
}