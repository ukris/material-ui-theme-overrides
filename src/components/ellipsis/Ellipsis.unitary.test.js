import React from 'react'
import renderer from 'react-test-renderer'
import { mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Ellipsis from './Ellipsis'

configure({ adapter: new Adapter() })


describe('Ellipsis | Non-editable', () => {
    const value = "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis vestibulum augue, ut ullamcorper velit. Etiam sodales odio ut luctus efficitur"
    const component = (
        <Ellipsis
            text={value}
        />
    )


    it('renders component correctly', () => {
        const wrapper = mount(component)
        const ellipsisContainer = wrapper.find('.table-ellipsis-container')
        const ellipsisValue = wrapper.find('input').props().value

        expect(ellipsisContainer.length).toEqual(1)
        expect(ellipsisValue).toEqual(value)
    });

    it('matches snapshot', () => {
        const tree = renderer.create(component,).toJSON()
        const ellipsisValue = tree.children[0].children[0].children[0].props.value

        expect(tree).toMatchSnapshot()
        expect(ellipsisValue).toEqual(value)
    });
})
