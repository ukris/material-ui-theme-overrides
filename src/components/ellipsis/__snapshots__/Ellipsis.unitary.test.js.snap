// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ellipsis | Non-editable matches snapshot 1`] = `
<div
  className="table-ellipsis-container"
>
  <div
    className="MuiFormControl-root MuiTextField-root table-ellipsis noneditable"
    data-test="table-ellipsis"
    title="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis vestibulum augue, ut ullamcorper velit. Etiam sodales odio ut luctus efficitur"
  >
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        id="standard-multiline-flexible"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque quis vestibulum augue, ut ullamcorper velit. Etiam sodales odio ut luctus efficitur"
      />
    </div>
  </div>
</div>
`;
